src/ObjWriting/ObjContainer/IWD/IWDWriter.cpp

```diff
@@ -0,0 +1,121 @@
+#include "IWDWriter.h"
+
+#include "../minizip/zip.h"
+
+#include <algorithm>
+#include <iostream>
+
+class IWDWriterImpl final : public IWDWriter
+{
+public:
+    explicit IWDWriterImpl(const std::string& iwdFileName, ISearchPath* assetSearchPath)
+        : m_filename(iwdFileName),
+          m_asset_search_path(assetSearchPath)
+    {
+        m_files.clear();
+        m_added_files.clear();
+        m_zipfile = zipOpen(m_filename.c_str(), APPEND_STATUS_CREATE);
+        if (!m_zipfile)
+        {
+            throw std::runtime_error("Error creating IWD file: \"" + m_filename + "\"");
+        }
+    }
+
+    void AddFile(std::string fileName) override
+    {
+        const auto existingEntry = std::find(m_added_files.cbegin(), m_added_files.cend(), fileName);
+        if (existingEntry != m_added_files.cend())
+        {
+            std::cerr << "Duplicate entry in " << m_filename << "'s zone definition; ignoring" << "\n";
+            return;
+        }
+        m_files.emplace_back(fileName);
+    }
+
+    void AddFileToArchive(const std::string& fileName, const std::unique_ptr<char[]>& fileData, const size_t& fileSize)
+    {
+        zip_fileinfo zi = {};
+
+        if (zipOpenNewFileInZip(m_zipfile, fileName.c_str(), &zi, NULL, 0, NULL, 0, NULL, Z_DEFLATED, Z_DEFAULT_COMPRESSION) != ZIP_OK)
+        {
+            zipClose(m_zipfile, NULL);
+            throw std::runtime_error("Could not open new file in IWD \"" + fileName + "\"\n");

```

Comment:
Please don't use `std::runtime_error` :sweat_smile: This does not log anything and will just crash and not graceful exit